.gitlab/build_ruby.yml

```diff
@@ -63,6 +63,14 @@ ruby-clang_10_0_0-src:
     HOST_CONFIG: "ruby-toss_3_x86_64_ib-${COMPILER}.cmake"
   extends: .src_build_on_ruby
 
+ruby-clang_10_0_0-release-src:
+  variables:
+    COMPILER: "clang@10.0.0"
+    HOST_CONFIG: "ruby-toss_3_x86_64_ib-${COMPILER}.cmake"
+    BUILD_TYPE: "Release"
+    EXTRA_CMAKE_OPTIONS: "-DAXOM_QUEST_ENABLE_EXTRA_REGRESSION_TESTS:BOOL=ON"
```

Comment:
Should options like this and `AXOM_ENABLE_MFEM_SIDRE_DATACOLLECTION` be documented in our [ReadTheDocs configuration options](https://axom.readthedocs.io/en/develop/docs/sphinx/quickstart_guide/config_build.html#cmake-configuration-options)?

Reply:
Thanks @bmhan12. In general, yes, we should document all our user-facing config options.

In the past, we have intentionally omitted a few niche options from our documentation, including the two that you mentioned:

* `AXOM_QUEST_ENABLE_EXTRA_REGRESSION_TESTS` is a developer option. We have a few quest regression tests that are always on and this option adds additional tests. 
* `AXOM_ENABLE_MFEM_SIDRE_DATACOLLECTION` was intended as a short-lived option -- it was added because we couldn't yet enable this feature by default, and we plan to remove this option (and enable the feature by default) as soon as we can. As far as I can tell, we should be able to do this once we have a working hip configuration.

There are a few other undocumented options in the code, like `AXOM_MINT_USE_SIDRE` that we might also want to consider
https://github.com/LLNL/axom/blob/3999cf27a5437b032f9d3e23ca4476c04d139a2e/src/axom/mint/CMakeLists.txt#L18-L19

Any of these should be easy enough to add to the docs. We might want to discuss this further as a team. I'll create an issue for it.